swri_roscpp/include/swri_roscpp/topic_service_client.h

```diff
@@ -0,0 +1,173 @@
+  {
+
+  }
+
+  void initialize(ros::NodeHandle &nh,
+                const std::string &service,
+                const std::string &client_name = "")
+  {
+    name_ = client_name.length() ? client_name : ros::this_node::getName();
+    std::string rservice = nh.resolveName(service);
+    service_name_ = rservice;
+    response_sub_ = nh.subscribe(rservice + "/response", 10, &TopicServiceClientRaw<MReq, MRes>::response_callback, this);
+    request_pub_ = nh.advertise<MReq>(rservice + "/request", 10);
+  }
+
+  bool call(MReq& request, MRes& response)
+  {
+    boost::mutex::scoped_lock scoped_lock(request_lock_);
+
+    // block for response
+    request.srv_header.stamp = ros::Time::now();
+    request.srv_header.sequence = sequence_;
+    request.srv_header.sender = name_;
+
+    // Wait until we get a subscriber and publisher
+    while (request_pub_.getNumSubscribers() == 0 && response_sub_.getNumPublishers() == 0)
+    {
+      ros::Duration(0.002).sleep();
+      ros::spinOnce();
+
+      if (ros::Time::now() - request.srv_header.stamp > timeout_)
+      {
+        return false;
+      }
+    }
+    request_pub_.publish(request);
+
+    // Wait until we get a response
+    while (!response_ && ros::Time::now() - request.srv_header.stamp < timeout_)
+    {
+      ros::Duration(0.002).sleep();
+      ros::spinOnce();
+    }
+
+    sequence_++;
+    if (response_)
+    {
+      response = *response_;
+      response_.reset();
+      return response.srv_header.result;
+    }
+    return false;
+  }
+
+  std::string getService()
+  {
+    return service_name_;
+  }
+
+  bool exists()
+  {
+    return request_pub_.getNumSubscribers() > 0;
+  }
+
+  //bool call_async(MReq& request, boost::function<void(const MReq&, const MRes&)>& fun);
+
+  // The service server can output a console log message when the
+  // service is called if desired.
+  void setLogCalls(bool enable);
+  bool logCalls() const;
+private:
+
+  void response_callback(const boost::shared_ptr<MRes>& message)
+  {
+    ROS_DEBUG("Got response for %s with sequence %i",
+             message->srv_header.sender.c_str(), message->srv_header.sequence);
+
+    if (message->srv_header.sender != name_)
+    {
+      ROS_DEBUG("Got response from another client, ignoring..");
+      return;
+    }
+
+    if (message->srv_header.sequence != sequence_)
+    {
+      ROS_WARN("Got wrong sequence number, ignoring..");
+      return;
+    }
+
+    response_ = message;
+  }
+};  // class TopicServiceClient
+
+template<class MReq>
+class TopicServiceClient: public TopicServiceClientRaw<typename MReq:: Request, typename MReq:: Response>
+{
+public:
+  bool call(MReq& req)
+  {
+    return TopicServiceClientRaw<typename MReq:: Request, typename MReq:: Response>::call(req.request, req.response);
+  }
+
+};  // class TopicServiceClientEz
+
+
+
+
+/*inline
+bool TopicServiceClient::call_async(MReq& request, boost::function<void(const MReq&, const MRes&)>& fun)
```

Comment:
Does this commented-out code need to be here?

Reply:
No, was just there in case I decided to add it. Just going to remove it.